src/twisted/internet/defer.py

```diff
@@ -515,54 +542,151 @@ def addCallbacks(
         #     the caller should treat the result as the new type, consistently.
         return cast(Deferred[_NextDeferredResultT], self)
 
+    @overload
+    def addCallback(
+        self,
+        callback: Callable[Concatenate[_DeferredResultT, _P], Failure],
+        *args: _P.args,
+        **kwargs: _P.kwargs,
+    ) -> Deferred[_NextDeferredResultT]:
+        ...
+
+    @overload
     def addCallback(
         self,
         callback: Callable[
-            ...,
-            "Union[_NextDeferredResultT, Deferred[_NextDeferredResultT]]",
+            Concatenate[_DeferredResultT, _P], Deferred[_NextDeferredResultT]
         ],
-        *args: object,
-        **kwargs: object,
-    ) -> "Deferred[_NextDeferredResultT]":
+        *args: _P.args,
+        **kwargs: _P.kwargs,
+    ) -> Deferred[_NextDeferredResultT]:
+        ...
+
+    @overload
+    def addCallback(
+        self,
+        callback: Callable[Concatenate[_DeferredResultT, _P], _NextDeferredResultT],
+        *args: _P.args,
+        **kwargs: _P.kwargs,
+    ) -> Deferred[_NextDeferredResultT]:
+        ...
+
+    def addCallback(
+        self,
+        callback: Any,
+        *args: Any,
+        **kwargs: Any,
+    ) -> "Deferred[Any]":
```

Comment:
Errrr :/  Those `Any` annotations look so sad.  PEP 484 and the Python library documentation both imply you can leave annotations _off_ of the final definition of an `@overload`ed function ... but I see that mypy complains about this function if we do so.

Is that because of something we set in our mypy config?  If it's possible to keep these `Any` annotations out it seems like that would be less confusing.  As far as I can tell, mypy does ignore these annotations (apart from requiring that they exist _at all_) so this is more about confusion to future maintainers and other humans.